tests/test_gin.py

```diff
@@ -138,5 +159,108 @@ class TestConverter(NWBConverter):
             check_sortings_equal(SX1=sorting, SX2=nwb_sorting)
 
 
+if HAVE_PARAMETERIZED and HAVE_OPHYS_DATA:
+
+    class TestOphysNwbConversions(unittest.TestCase):
+        savedir = Path(tempfile.mkdtemp())
+        imaging_interface_list = []#debug
+
+        # imaging_interface_list = [
+        #     param(
+        #         data_interface=TiffImagingInterface,
+        #         interface_kwargs=dict(file_path=str(OPHYS_DATA_PATH / "imaging_datasets" / "Tif" / "demoMovie.tif")),
+        #     ),
+        #     param(
+        #         data_interface=Hdf5ImagingInterface,
+        #         interface_kwargs=dict(file_path=str(OPHYS_DATA_PATH / "imaging_datasets" / "hdf5" / "demoMovie.hdf5")),
+        #     ),
+        # ]
+        for suffix in [".mat", ".sbx"]:
+            imaging_interface_list.append(
+                param(
+                    data_interface=SbxImagingInterface,
+                    interface_kwargs=dict(
+                        file_path=str(OPHYS_DATA_PATH / "imaging_datasets" / "Scanbox" / f"sample{suffix}")
+                    ),
+                ),
+            )
+
+        @parameterized.expand(imaging_interface_list, name_func=custom_name_func)
+        def test_convert_imaging_extractor_to_nwb(self, data_interface, interface_kwargs):
+            nwbfile_path = str(self.savedir / f"{data_interface.__name__}.nwb")
+
+            class TestConverter(NWBConverter):
+                data_interface_classes = dict(TestImaging=data_interface)
+                def get_metadata(self):
+                    metadata = super().get_metadata()
+                    # attach device to ImagingPlane lacking property
+                    # assume single device and single imaging plane
+                    if 'device' not in metadata['Ophys']['ImagingPlane'][0].keys():
+                        metadata['Ophys']['ImagingPlane'][0]['device'] = metadata['Ophys']['Device']
```

Comment:
@CodyCBakerPhD the schema requires the 'device' property an `ImagingPlane` to be a string, assigning the device itself like I do here leads to an error. What's the proper way to do this here? Is there something like softlinks in matnwb that will figure out the path to the device by itself?

Reply:
In metadata, all values are simply strings. In this case, the string name of the device for that object, and hopefully the imaging write function (which should take metadata as an argument) will then handle the actual object referencing in the NWBFile based on that string name.

Reply:
Because we don't have great examples of, and little work done using `ophys` - I'd recommend seeing how it's done in `ecephys` for things we rigorously test already: for example, https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py#L70-L77

Reply:
perfect. I was looking for something like this